src/frontend/screens/MapScreen/MapLayers/RemoteDetectionAlertsLayer.tsx

```diff
@@ -0,0 +1,129 @@
+  const {data: alerts} = useRemoteDectionAlerts();
+
+  if (!alerts) {
+    return null;
+  }
+
+  return (
+    <MapboxGL.ShapeSource
+      id="alerts-source"
+      shape={convertRemoteDetectionAlertsToFeatures(alerts)}>
+      {/* Symbol Layer for Labels */}
+      <MapboxGL.SymbolLayer
+        id="mapeo-alerts-label"
+        filter={[
+          'all',
+          [
+            'in',
+            '$type',
+            'Polygon',
+            'LineString',
+            'Point',
+            'MultiLineString',
+            'MultiPolygon',
+            'MultiPoint',
+          ],
+          ['has', 'metadata.alert_type'],
+          ['has', 'month_detec'],
+          ['has', 'year_detec'],
+        ]}
+        style={{
+          textField: [
+            'concat',
+            ['get', 'metadata.alert_type'],
+            ' (',
+            ['get', 'month_detec'],
+            '-',
+            ['get', 'year_detec'],
+            ')',
+          ],
+          textFont: ['Open Sans Regular'],
+          textOffset: [0, 0.5],
+          textAnchor: 'top',
+          textColor: '#FFFFFF',
+          textHaloColor: '#000000',
+          textHaloWidth: 1,
+          textHaloBlur: 1,
+        }}
+      />
+
+      {/* Circle Layer for Points */}
+      <MapboxGL.CircleLayer
+        id="comapeo-alerts-point"
+        filter={['==', '$type', 'Point', 'MultiPoint']}
+        style={{
+          circleRadius: 5,
+          circleColor: '#FF0000',
+        }}
+      />
+
+      {/* Line Layer for LineStrings and MultiLineStrings */}
+      <MapboxGL.LineLayer
+        id="comapeo-alerts-linestring"
+        filter={['in', '$type', 'LineString', 'MultiLineString']}
+        style={{
+          lineColor: '#FF0000',
+          lineWidth: 3,
+          lineOpacity: 0.8,
+        }}
+      />
+
+      {/* Line Layer for Polygon Stroke */}
+      <MapboxGL.LineLayer
+        id="comapeo-alerts-polygon-stroke"
+        filter={['in', '$type', 'Polygon', 'MultiPolygon']}
+        style={{
+          lineColor: '#FF0000',
+          lineWidth: 2,
+        }}
+      />
+
+      {/* Fill Layer for Polygon Fill */}
+      <MapboxGL.FillLayer
+        id="comapeo-alerts-polygon"
+        filter={['in', '$type', 'Polygon', 'MultiPolygon']}
+        style={{
+          fillColor: '#FF0000',
+          fillOpacity: 0.5,
+        }}
+      />
+    </MapboxGL.ShapeSource>
+  );
+};
+
+function convertRemoteDetectionAlertsToFeatures(
+  alerts: RemoteDetectionAlert[],
+): FeatureCollection {
+  return {
+    type: 'FeatureCollection',
+    features: alerts.map(alert => {
+      const dateStart = new Date(alert.detectionDateStart);
+      return {
+        type: 'Feature',
+        geometry: alert.geometry,
+        properties: {
+          ...flatten(
+            includeKeys(alert, [
+              'metadata',
+              'detectionDateStart',
+              'detectionDateEnd',
+              'sourceId',
+            ]),
+          ),
```

Comment:
what's the desired outcome here? This feels prone to "overflattening" and I would suggest using explicit fields where flattening isn't needed:

```
          metadata: alert.metadata, // Maybe this one needs flattening? Not sure what the desired outcome is...
          detectionDateStart: alert.detectionDateStart,
          detectionDateEnd: alert.detectionDateEnd,
          sourceId: alert.sourceId,
```

Doing the above potentially avoids needing to install deps too (for now)

Reply:
my guess is that it's related to the mapbox filter expressions, in which case I would suggest just lifting the relevant `metadata` fields explicitly such that they're in the `properties` geojson field e.g.

```ts
properties: {
  alertType: metadata.alertType,
  ...
}
```

and update the filter expression to be `['get', 'alertType']`
  
There seems to be a way to write a mapbox style expression to query nested fields in a object `properties` field but I think it's not necessary to try it.

Reply:
I believe (but im not 100% sure) that there can be certain properties in metadata (including `alert_type`) that is used by `properties`. So this flattening allows us to access this.

Gregor and I discussed this, and we came to this code (using the deps)

Your way only allows for the `alert_type` to be extracted. But maybe that ok? Im not sure if there are other properties that are needed in metadata?

Reply:
> Your way only allows for the alert_type to be extracted. But maybe that ok? Im not sure if there are other properties that are needed in metadata?

Understandable, just feels like a premature optimization for something that can be implemented with much less code - an example of accounting for something that isn't needed yet

Reply:
seems like the only code that's using the properties are the layer filters, which so far only needs the `alert_type` from `metadata`. Would make more sense to make more of the metadata fields accessible in `properties` if - for example - the layer was pressable and some interaction needed access to those fields via the geojson properties

Reply:
alertType: metadata.alert_type